code/datums/components/dusting.dm

```diff
@@ -0,0 +1,249 @@
+			span_hear("You hear a loud crack as you are washed with a wave of heat."))
+		qdel(consumed_object)
+	if(isturf(consumed_atom))
+		if(isspaceturf(consumed_atom)) //don't scrape away reality at its seams
+			return
+		var/turf/old_turf = consumed_atom
+		var/old_turf_name = old_turf.name
+		var/turf/old_turf_type = old_turf.type
+		var/turf/new_turf = old_turf.ScrapeAway(flags = CHANGETURF_INHERIT_AIR)
+		if(new_turf.type == old_turf_type)
+			return
+		new_turf.visible_message(span_danger("[atom_parent] consumes \the [old_turf_name] and reveals [new_turf] underneath."), null,
+			span_hear("You hear a loud crack as you are washed with a wave of heat."))
+	playsound(get_turf(parent), consumption_sound, 80, vary = TRUE)
+
+	//Next, do radiation and give fluff text to nearby mobs
+	. = TRUE
+
+	var/radiation_array = "default"
+	if(radiation_atom[atom_parent.type])
+		radiation_array = atom_parent.type
+	else
+		var/actual_subtype //used to check the best subtype in the list without breaking early. if we broke early, obj/item/sword would return an obj value when obj/item/sword is also defined
+		for(var/typepath in radiation_atom)
+			if(istype(atom_parent, typepath) && (ispath(typepath, actual_subtype) || !actual_subtype))
+				radiation_array = typepath
+				actual_subtype = typepath
+
+	var/range = radiation_atom[radiation_array]["range"]
+	var/threshold = radiation_atom[radiation_array]["threshold"]
+	var/chance = radiation_atom[radiation_array]["chance"]
+
+	if(!range || !threshold || !chance && !(!range && !threshold && !chance)) //Don't continue w/ radiation nor its messages if any of these are set to 0, but not when all are zero
+		stack_trace("Incorrect radiation_pulse arguments (range=[range], threshold=[threshold], chance=[chance]).") //Send a warning just in case this was never intended
+		return
+	radiation_pulse(parent, max_range = range, threshold = threshold, chance = chance)
+	var/visual_message = rename_text(consume_text_visual, consumed_atom)
+	var/audible_message = rename_text(consume_text_audible, consumed_atom)
+	var/unseen_message = rename_text(consume_text_unseen, consumed_atom)
+	for(var/mob/living/near_mob in range(consume_text_range, parent))
+		if(HAS_TRAIT(near_mob, TRAIT_RADIMMUNE) || issilicon(near_mob))
+			continue
+		if(ishuman(near_mob) && SSradiation.wearing_rad_protected_clothing(near_mob))
+			continue
+		if(near_mob in view(consume_text_range, atom_parent))
+			near_mob.show_message(visual_message, MSG_VISUAL, audible_message, MSG_AUDIBLE)
+		else
+			near_mob.show_message(unseen_message, MSG_AUDIBLE)
+		atom_parent.investigate_log("has irradiated [key_name(near_mob)] after consuming [consumed_atom].", INVESTIGATE_SUPERMATTER)
+
+//Helper proc
+/datum/component/dusting/proc/rename_text(text, target)
+	. = replacetext(text, "%parent%", "[atom_parent]")
+	if(target)
+		return replacetext(., "%target%", "\The [target]")
```

Comment:
What is this for?

Reply:
> You're using GetComponent a lot, which is never a great sign. A lot of this seems like very supermatter specific behavior, I'm not sure it's a great thing to generalize. Is this just for admin memes?

Mostly for admemes, yeah. I dont really see where future code will use this but hey, we have organs that make you puke slimes, at some point this component will be used outside admemery and the sm sword